### PR TITLE
fix: remove obsoleted 'version' elements in compose files

### PR DESCRIPTION
**Pull request by kurokobo**
# Description

This PR removes the line `version: '3'` from all `docker-compose*.yaml` files.

The `version` element is already obsoleted, and causes warning to be shown in the newer compose plugin:

> The top-level version property is defined by the Compose Specification for backward compatibility. It is only informative and you'll receive a warning message that it is obsolete if used.
> https://docs.docker.com/compose/compose-file/04-version-and-name/

Example warnings are here; the same warning will be shown on any other `docker compose` commands:

```bash
$ docker compose up -d
WARN[0000] /home/********/dify/docker/docker-compose.yaml: `version` is obsolete      👈👈👈👈👈
[+] Running 12/12
 ✔ Network docker_ssrf_proxy_network  Created                                                                                                                                                                   0.1s 
 ✔ Network docker_default             Created                                                                                                                                                                   0.1s 
 ✔ Container docker-sandbox-1         Started                                                                                                                                                                   0.5s 
 ✔ Container docker-weaviate-1        Started                                                                                                                                                                   0.8s 
 ✔ Container docker-web-1             Started                                                                                                                                                                   0.6s 
 ✔ Container docker-db-1              Started                                                                                                                                                                   0.7s 
 ✔ Container docker-ssrf_proxy-1      Started                                                                                                                                                                   0.9s 
 ✔ Container docker-redis-1           Started                                                                                                                                                                   0.6s 
 ✔ Container docker-bot-1             Started                                                                                                                                                                   0.7s 
 ✔ Container docker-worker-1          Started                                                                                                                                                                   1.5s 
 ✔ Container docker-api-1             Started                                                                                                                                                                   1.4s 
 ✔ Container docker-nginx-1           Started        

$ docker compose logs api --tail=5
WARN[0000] /home/********/dify/docker/docker-compose.yaml: `version` is obsolete       👈👈👈👈👈
api-1  | None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
api-1  | [2024-06-24 12:57:14 +0000] [22] [INFO] Starting gunicorn 22.0.0
api-1  | [2024-06-24 12:57:14 +0000] [22] [INFO] Listening at: http://0.0.0.0:5001 (22)
api-1  | [2024-06-24 12:57:14 +0000] [22] [INFO] Using worker: gevent
api-1  | [2024-06-24 12:57:14 +0000] [37] [INFO] Booting worker with pid: 37

$ docker compose exec api flask db history
WARN[0000] /home/********/dify/docker/docker-compose.yaml: `version` is obsolete        👈👈👈👈👈
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
4e99a8df00ff -> 7b45942e39bb (head), add-api-key-auth-binding
64a70a7aab8b -> 4e99a8df00ff, add load balancing
03f98355ba0e -> 64a70a7aab8b, add workflow run index
9e98fbaffb88 -> 03f98355ba0e, add workflow tool label and tool bindings idx
.....
```

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tried invoking `docker compose` commands such as `down`, `up`, `logs`, `ps`, `pull`, `start`, `stop` and confirmed that there is no warning

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by takatost** [comment]
Hey, I couldn't find in the docker doc which specific version of the `docker compose` client started supporting the loose version parameter. If we remove it abruptly, it might affect existing users. Do you have more information on this?

**Comment by kurokobo** [comment]
@takatost 
Hi, thanks for your review!

> which specific version of the `docker compose` client started supporting the loose version parameter.

**TL;DR**
- Compose stopped referencing `version` about **4 years ago**, Sep. 2020,  back when everyone was still using the `docker-compose` command instead of `docker compose` command.
- By the time the `compose` subcommand was added to the `docker` command, `version` was already obsolete. This was in 2022.
- The old Compose V1 engine that used `version` already reached EOL in Jun. 2023.

**Details:**
Docker Compose stopped using `version` with the release of [Docker Compose 1.27.0](https://docs.docker.com/compose/release-notes/#1270) on 2020-09-07. Up until version 1.26.x, Docker Compose used _versioned_ schemas like 2.x and 3.x, but these were unified into a single _non-versioned_ schema known as the _Compose Specification_. Docker Compose started using this specification from version 1.27.0 onwards, where `version` became merely informative.

> Legacy versions 2.x and 3.x of the Compose file format were merged into the Compose Specification. It is implemented in versions 1.27.0 and above (also known as Compose V2) of the Docker Compose CLI.
> https://docs.docker.com/compose/compose-file/

> The top-level version property is defined by the Compose Specification for backward compatibility. It is only informative
> https://docs.docker.com/compose/compose-file/04-version-and-name/

Just before the release of 1.27.0, [a PR](https://github.com/compose-spec/compose-spec/pull/82/files) makes `version` to optional, and around the same time, [another PR](https://github.com/docker/compose/pull/7588/files) removed all versioned schemas and unified them into one schema without a version number. 

The addition of the `compose` subcommand to the `docker` command happened two years later in April 2022 with the release of Compose V2, which also (and of course) uses the Compose Specification and thus does not recognize `version`.

As of today, `version` only matters if the users are using an old Compose V1 (pre-1.27) with the traditional `docker-compose` command. And it has been announced that Compose V1 will reach EOL in June 2023 (see [Proposed Compose V1 End-of-Life Timeline on this page](https://www.docker.com/blog/announcing-compose-v2-general-availability/)).

**Summary:**
From my understanding, `version` is meaningless in most current environments and can be safely removed.
If there is any user impact due to this change, it means those users have been running an outdated Docker Compose for at least 4 years without upgrading.
I believe there is little value in maintaining support for such EOL-ed environments at the expense of new environment users receiving `WARN`s.

**Review by takatost — APPROVED** [review]
thanks for your kind explanation!
